api/src/database/helpers/schema/dialects/mssql.ts

```diff
@@ -2,6 +2,16 @@ import { Knex } from 'knex';
 import { SchemaHelper } from '../types';
 
 export class SchemaHelperMSSQL extends SchemaHelper {
+	applyLimit(rootQuery: Knex.QueryBuilder, limit: number): void {
+		// The ORDER BY clause is invalid in views, inline functions, derived tables, subqueries,
+		// and common table expressions, unless TOP, OFFSET or FOR XML is also specified.
+		if (limit === -1) {
+			rootQuery.limit(Number.MAX_SAFE_INTEGER);
```

Comment:
Do we need to set a limit here? I believe a SQL query without explicit limit should return all items right? or am i missing a MS SQL edge case 🤔

Reply:
Yes we do, in order to allow for sorting in a subquery.
Technically we should be using `TOP (100) percent`, but it's not supported when using limits from Knex. So the MAX_SAFE_INTEGER is used.

Refer to the two comment lines prior.
https://github.com/directus/directus/blob/7b8353cb3d9b94022de733c2ddee933894b15f40/api/src/database/helpers/schema/dialects/mssql.ts#L6-L7